client/component.go

```diff
@@ -1,23 +1,107 @@
+
+// Copyright 2023 PANTHEON.tech
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package client
 
-import "go.pantheon.tech/stonework/proto/cnfreg"
+import (
+	"context"
+	"fmt"
+
+	"go.ligato.io/vpp-agent/v3/cmd/agentctl/api/types"
+	"go.ligato.io/vpp-agent/v3/cmd/agentctl/client"
+
+	"go.ligato.io/vpp-agent/v3/proto/ligato/kvscheduler"
+	"go.pantheon.tech/stonework/plugins/cnfreg"
+	cnfregpb "go.pantheon.tech/stonework/proto/cnfreg"
+)
+
+type ComponentMode int32
+
+const (
+	// Foreign means the component is not managed by StoneWork
+	ComponentForeign ComponentMode = iota
+
+	// StoneworkModule means the component is a StoneWork module managed by StoneWork
+	ComponentStoneworkModule
+
+	// Stonework means the component is a StoneWork instance
+	ComponentStonework
+)
 
 // Component is a component of StoneWork. It can be StoneWork instance itself,
 // a CNF connected to it or other Ligato service in connected to StoneWork.
 type Component interface {
-	Name() string
-	Mode() cnfreg.CnfMode
+	GetName() string
+	GetMode() ComponentMode
+	GetInfo() *cnfreg.Info
+	GetMetadata() map[string]string
+	SchedulerValues() ([]*kvscheduler.BaseValueStatus, error)
 }
 
 type component struct {
-	name string
-	mode cnfreg.CnfMode
+	agentclient *client.Client
+	Name        string
+	Mode        ComponentMode
+	Info        *cnfreg.Info
+	Metadata    map[string]string
+}
+
+func (c *component) GetName() string {
+	return c.Name
 }
 
-func (c *component) Name() string {
-	return c.name
+func (c *component) GetMode() ComponentMode {
+	return c.Mode
+}
+
+func (c *component) GetInfo() *cnfreg.Info {
+	return c.Info
+}
+
+func (c *component) Client() *client.Client {
+	return c.agentclient
+}
+
+func (c *component) GetMetadata() map[string]string {
+	return c.Metadata
+}
+
+func (c *component) SchedulerValues() ([]*kvscheduler.BaseValueStatus, error) {
+	if c.Mode == ComponentForeign {
+		return nil, fmt.Errorf("cannot get scheduler values of component %s, this component in not managed by StoneWork", c.Name)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	values, err := c.agentclient.SchedulerValues(ctx, types.SchedulerValuesOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return values, nil
 }
 
-func (c *component) Mode() cnfreg.CnfMode {
-	return c.mode
+func cnfModeToCompoMode(cm cnfregpb.CnfMode) ComponentMode {
+	switch cm {
+	case cnfregpb.CnfMode_STANDALONE:
+		return ComponentForeign
```

Comment:
I do not that standalone should be considered as foreign component.

Reply:
Foreign was meant to mean unmanaged by StoneWork service and standalone CNFs are not managed by StoneWork. But to be more clear about this I added `ComponentStandalone` to represent standalone CNFs and foreign is now called auxiliary (thes are neither CNFs nor StoneWork).